From: Petr Mladek <pmladek@suse.cz>
Date: Wed, 16 Apr 2014 17:19:44 +0200
Subject: [RFC PATCH 10/11] printk: survive rotation of sequence numbers
Patch-mainline: Submitted https://lkml.org/lkml/2014/5/9/145
References: bnc#831949

NMI log buffer uses half of "unsigned long" to store the sequence number.
The maximum value might be relatively small, especially on 32-bit systems.
This patch makes the check for free space more strict, so it does give
false results after the rotation.

Signed-off-by: Petr Mladek <pmladek@suse.cz>
---
 kernel/printk/printk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/printk/printk.c b/kernel/printk/printk.c
index 51dd26606727..6e006ce248cb 100644
--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -578,7 +578,7 @@ static int log_make_free_space(enum printk_log_type log, u32 msg_size)
 	int freed = 0;
 	int ret = 0;
 
-	while (get_first_seq(log) < get_next_seq(log)) {
+	while (get_first_seq(log) != get_next_seq(log)) {
 		if (logbuf_has_space(log, msg_size, false))
 			goto out;
 
-- 
1.8.4

